### PR TITLE
remove wrong shim thickness and typo for top back UChannel

### PR DESCRIPTION
**Pull request by sarahgaiser**
Removes 700um shift that Cam notices during his Survey Constant analysis and fixes bug in geometry code. The bug was as follows: by not setting the node for the top back alignment corrections, the corresponding survey volume had no access to the survey constants defined in the `compact.xml`. Fixing the typo solves this problem.

### CODE REVIEW

**Review by cbravo135 — APPROVED** [review]
I checked out this code and made comparisons between the 2016 and 2019 detectors and confirmed this does indeed fix the issue of getting different detectors with the same survey data.

File: detector-model/src/main/java/org/lcsim/geometry/compact/converter/HPSTracker2019GeometryDefinition.java [hunk 1]
```diff
@@ -556,7 +556,7 @@ public static class ModuleL3Bot extends ModuleL13Bot {
         //2019 MRSolt survey
         //protected final static double L3_new_vertical_shift = 0.7 - 0.012;
         //2021 nominal
-        protected final static double L3_new_vertical_shift = 0.7;
+        protected final static double L3_new_vertical_shift = 0.;
```
Comment (pbutti) [inline detector-model/src/main/java/org/lcsim/geometry/compact/converter/HPSTracker2019GeometryDefinition.java:559]: Any clue why Matt Added this 0.7mm. ?

File: detector-model/src/main/java/org/lcsim/geometry/compact/converter/HPSTracker2019GeometryDefinition.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ public void build() {
         AlignmentCorrection supBotCorrBack = getUChannelCorrection(false,90);
         supBotCorrBack.setNode(node);
         AlignmentCorrection supTopCorrBack = getUChannelCorrection(true,90);
-        supTopCorr.setNode(node);
+        supTopCorrBack.setNode(node);
```
Comment (pbutti) [inline detector-model/src/main/java/org/lcsim/geometry/compact/converter/HPSTracker2019GeometryDefinition.java:54]: Indeed a bug!

**Review by pbutti — APPROVED** [review]

File: detector-model/src/main/java/org/lcsim/geometry/compact/converter/HPSTracker2019GeometryDefinition.java (hunk 1 above)
Comment (cbravo135) [reply detector-model/src/main/java/org/lcsim/geometry/compact/converter/HPSTracker2019GeometryDefinition.java:559]: My best guess is this was to account for the shims added wrt the engineering run detector. I don't think it was that thick, will confirm with Tim. Also, this isn't even used if you supply survey data for the modules.

**Review by cbravo135 — CHANGES_REQUESTED** [review]
Need to properly update detector lcdd files.

**Comment by normangraf** [comment]
Please hold off on merging this into master until we can discuss this at the software meeting. I would much prefer that we maintain backwards-compatibility with four years of simulation and reconstruction work. Changing the geometry in this way will invalidate those detectors. We should modify the compact.xml files to be commensurate with the new code such that the output geometry is unchanged. We should then proceed from here on out with new detectors that correct the mistake.

**Review by cbravo135 — APPROVED** [review]
We have not done much simulation/reconstruction work in the past 4 years. If we need to rereco some MC for some reason we can always use the 5.1 release to have the same detectors. Making things "backwards compatible" to be able to keep a bug around is poor software practice.

**Pull request merged**